src/DGtal/geometry/surfaces/estimation/PlaneProbingLNeighborhood.h

```diff
@@ -0,0 +1,300 @@
+       */
+      ClosestGridPoint (const GridPoint& aGridPoint,
+			const bool& aFirst, const bool& aSecond )
+	: myGridPoint(aGridPoint), myPair(std::make_pair(aFirst,aSecond)) {}
+
+      GridPoint myGridPoint; /**< a grid point, which can be invalid 
+				 if no grid point belong to the underlying surface */
+
+      std::pair<bool,bool> myPair; /**< pair of boolean values that encode 
+				       the local configuration */
+    };
+    
+    // ----------------------- Standard services ------------------------------
+  public:
+    /**
+     * Default constructor.
+     */
+    PlaneProbingLNeighborhood() = delete;
+
+    /**
+     * Constructor.
+     *
+     * @param aPredicate a probing predicate.
+     * @param aQ the fixed point 'q'.
+     * @param aM a frame composed of the three vectors.
+     */
+    PlaneProbingLNeighborhood(Predicate const& aPredicate, Point const& aQ, Triangle const& aM);
+
+    /**
+     * Destructor.
+     */
+    ~PlaneProbingLNeighborhood();
+
+    /**
+     * Copy constructor.
+     * @param other the object to clone.
+     */
+    PlaneProbingLNeighborhood ( const PlaneProbingLNeighborhood & other ) = delete;
+
+    /**
+     * Move constructor.
+     * @param other the object to move.
+     */
+    PlaneProbingLNeighborhood ( PlaneProbingLNeighborhood && other ) = delete;
+
+    /**
+     * Copy assignment operator.
+     * @param other the object to copy.
+     * @return a reference on 'this'.
+     */
+    PlaneProbingLNeighborhood & operator= ( const PlaneProbingLNeighborhood & other ) = delete;
+
+    /**
+     * Move assignment operator.
+     * @param other the object to move.
+     * @return a reference on 'this'.
+     */
+    PlaneProbingLNeighborhood & operator= ( PlaneProbingLNeighborhood && other ) = delete;
+
+    // ----------------------- Plane-Probing services ------------------------------
+  public:
+    
+    /**
+     * Computes the current state of the neighborhood.
+     * This is the function that is overloaded for the different probing modes.
+     *
+     * @return the hexagon state, see HexagonState.
+     */
+    HexagonState hexagonState () override;
+
+    /**
+     * Computes the closest candidate point, used for updating a frame in a plane probing based estimator.
+     *
+     * @return the update operation to apply.
+     */
+    UpdateOperation closestCandidate () override;
+    
+    // ------------------------- Protected Datas ------------------------------
+  protected:
+
+    std::vector<ClosestGridPoint> myGrids; /**< closest point and additional useful data stored at each vertex. */
+    
+    // ------------------------- Helpers to find a closest point --------------
+  protected:
+    
+    /**
+     * Computes the closest candidate point in a given grid identified by the 
+     * index of the associated vertex.
+     *
+     * @param aIdx
+     * @return an instance of ClosestGridPoint.
+     */
+    ClosestGridPoint closestInGrid (const int& aIdx) const;
```

Comment:
int → size_t  or some index type of the container ?

Reply:
Since the last commit, an index type is used. Note however that 
- several classes have been impacted for consistency 
- and a method name has been changed in order to avoid any ambiguity.  ('index' to 'position' in 'PointOnProbingRay'; it is an integer that locates a point along a discrete ray)